File: A8/main.py
"""
A8. Dots naturals skaitlis n. Izdrukat tos skaitla n reizinatajus,
kuri ir kada naturala skaitla kvadrati.
"""

ok = 1
while ok==1:
    while True:
        n = int(input("Ievadi ne nulles naturalu skaitli N: "))
        if n<1:
            print("N ir jabut ne nulles naturalam skaitlim! Megini velreiz")
            continue
        else: break
    for i in range(1, n+1):
        if(n%i==0):
            for j in range(1, i+1):
                if j*j==i:
                    print(f"Reizinatajs {i} ir {j}^2")
    ok = int(input("Turpinat (1) vai Beigt (0): "))

File: A8/main.cpp
#include <iostream>
using namespace std;

/*
A8. Dots naturals skaitlis n. Izdrukat tos skaitla n reizinatajus,
kuri ir kada naturala skaitla kvadrati.
*/

int main(){
   int n, ok;
   do{
        while(true){
            cout<<"Ievadi ne nulles naturalu skaitli N: ";
            cin>>n;
            if(n<1){
                cout<<"N ir jabut ne nulles naturalam skaitlim! Megini velreiz" <<endl;
                continue;
            }else break;
        }
        for(int i=1; i<=n; i++){
            if(n%i==0){
                for(int j=1; j<=i; j++){
                    if(j*j==i){
                        cout<<"Reizinatajs " <<i <<" ir " <<j <<"^2" <<endl;
                    }
                }
            }
        }
    cout<<"Turpinat (1) vai Beigt (0): ";
    cin>>ok;
   }while(ok==1);
}
